### PR TITLE
 api status checker

### PR DESCRIPTION
**Pull request by CatalinFaurPaul**
## Proposed Changes
-  Add Elastic (count) metrics
-  Add Gateway current rounds metrics for every shard

## How to test
- when statusChecker flag is active elastic metrics and gateway metrics should work

### CODE REVIEW

**Comment by tanghel** [comment]
todo:
- total identities
- total nodes
- total providers
- total rounds
- total shards
- total mex pairs (and create `/count` endpoint)
- total mex farms (and create `/count` endpoint)
- total mex tokens (and create `/count` endpoint)
- create one histogram for all economics values (`economicsHistogram` / `setEconomicsValue(name, value)`)
- create one histogram for all maiar exchange values (`mexEconomicsHistogram` / `setMexEconomicsValue(name, value)`)

File: src/crons/status.checker/status.checker.service.ts [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_SECONDS)
+  async handleTokensSupplyInvalidations() {
+    await Locker.lock('Tokens supply invalidations', async () => {
+      await this.lock.acquire('tokens supply', async () => {
+        const performanceProfiler = new PerformanceProfiler();
+        const tokens = await this.tokenService.getTokens(new QueryPagination({ size: 1000 }), new TokenFilter());
```
Comment (gabrielmatei) [inline src/crons/status.checker/status.checker.service.ts:296]: why are we only checking the first 1000 tokens?

File: src/crons/status.checker/status.checker.service.ts [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async handleTokensAssetsInvalidations() {
+    await Locker.lock('Tokens assets', async () => {
+      await this.lock.acquire('assets', async () => {
+        const performanceProfiler = new PerformanceProfiler();
+        const tokens = await this.tokenService.getTokens(new QueryPagination({ size: 1000 }), new TokenFilter());
```
Comment (gabrielmatei) [inline src/crons/status.checker/status.checker.service.ts:317]: same here

File: src/crons/status.checker/status.checker.service.ts [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async handleTokensAccountsInvalidations() {
+    await Locker.lock('Tokens accounts ', async () => {
+      await this.lock.acquire('accounts', async () => {
+        const performanceProfiler = new PerformanceProfiler();
+        const tokens = await this.tokenService.getTokens(new QueryPagination({ size: 1000 }), new TokenFilter());
```
Comment (gabrielmatei) [inline src/crons/status.checker/status.checker.service.ts:337]: same here

File: src/crons/status.checker/status.checker.service.ts [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+      });
+    }, true);
+  }
+
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  async handleTokensTransactionsInvalidations() {
+    await Locker.lock('Tokens transactions ', async () => {
+      await this.lock.acquire('transactions', async () => {
+        const performanceProfiler = new PerformanceProfiler();
+        const tokens = await this.tokenService.getTokens(new QueryPagination({ size: 1000 }), new TokenFilter());
```
Comment (gabrielmatei) [inline src/crons/status.checker/status.checker.service.ts:357]: same here

File: src/crons/status.checker/status.checker.service.ts (hunk 1 above)
Comment (CatalinFaurPaul) [reply src/crons/status.checker/status.checker.service.ts:296]: Already discussed

File: src/crons/status.checker/status.checker.service.ts (hunk 3 above)
Comment (CatalinFaurPaul) [reply src/crons/status.checker/status.checker.service.ts:337]: Already discussed

File: src/crons/status.checker/status.checker.service.ts (hunk 4 above)
Comment (CatalinFaurPaul) [reply src/crons/status.checker/status.checker.service.ts:357]: Already discussed

**Review by tanghel — APPROVED** [review]

**Pull request merged**